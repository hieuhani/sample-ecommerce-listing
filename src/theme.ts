import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
  colors: {
    reddishOrange: '#f5471e',
    coolGrey: '#8f9598',
    silver: '#c6cace',
    tomato: '#ea341f',
    darkGrey: '#262829',
    paleGrey: '#eef1f3',
    white: '#FFF',
    black30: 'rgba(0, 0, 0, .3)',
    pumpkinOrange: '#f58400',
    deepSkyBlue: '#1e75f6',
  }
}
