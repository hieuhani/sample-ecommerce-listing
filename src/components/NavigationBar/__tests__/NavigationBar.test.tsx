import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { NavigationBar } from '..'
import { theme } from '../../../theme'

jest.mock('../../../contexts/navigation', () => ({
  useNavigationDispatch: jest.fn(() => null)
}))

describe('NavigationBar', () => {
  test('NavigationBar renders correctly', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <NavigationBar>
          Children
        </NavigationBar>
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('NavigationBar with primary variant', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <NavigationBar variant="primary">
          Children
        </NavigationBar>
      </ThemeProvider>
    )
    expect(container.querySelector('img')).toHaveAttribute('src', '/static/images/ArrowBackWhite.png')
  })

  test('NavigationBar with primary secondary', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <NavigationBar variant="secondary">
          Children
        </NavigationBar>
      </ThemeProvider>
    )
    expect(container.querySelector('img')).toHaveAttribute('src', '/static/images/ArrowBackBlack.png')
  })
})
