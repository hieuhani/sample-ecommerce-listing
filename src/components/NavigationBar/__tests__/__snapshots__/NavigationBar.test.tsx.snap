// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar NavigationBar renders correctly 1`] = `
.c0 {
  height: 44pt;
  background: rgb(232,32,32);
  background: linear-gradient(90deg,rgba(232,32,32,1) 0%,#f5471e 100%);
  border-bottom: 1px solid #eef1f3;
  padding: 8pt 11pt;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 24pt;
  height: 24pt;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 11pt;
  border: none;
  background-color: transparent;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    <img
      src="/static/images/ArrowBackWhite.png"
      srcSet="/static/images/ArrowBackWhite@2x.png 2x, /static/images/ArrowBackWhite@3x.png 3x"
    />
  </button>
  Children
</div>
`;
